src/components/AppLayout/InvalidMasterCopyError/index.tsx

```diff
@@ -0,0 +1,55 @@
+import { Link } from '@gnosis.pm/safe-react-components'
+import { useSelector } from 'react-redux'
+import { getChainInfo } from 'src/config'
+import { Errors, logError } from 'src/logic/exceptions/CodedException'
+import useAsync from 'src/logic/hooks/useAsync'
+import { currentSafe } from 'src/logic/safe/store/selectors'
+import { isValidMasterCopy } from 'src/logic/safe/utils/safeVersion'
+import MuiAlert from '@material-ui/lab/Alert'
+import MuiAlertTitle from '@material-ui/lab/AlertTitle'
+import { withStyles } from '@material-ui/core'
+
+const CLI_LINK = 'https://github.com/5afe/safe-cli'
+
+export const InvalidMasterCopyError = ({ onClose }: { onClose: () => void }): React.ReactElement | null => {
+  const chainInfo = getChainInfo()
+  const { currentVersion, address } = useSelector(currentSafe)
+
+  const [validMasterCopy, error] = useAsync(async () => {
+    if (address && currentVersion) {
+      return await isValidMasterCopy(chainInfo.chainId, address, currentVersion)
+    }
+  }, [address, chainInfo, currentVersion])
+
+  if (error) {
+    logError(Errors._620, error.message)
+    return null
+  }
+
+  if (typeof validMasterCopy === 'undefined' || validMasterCopy) {
+    return null
+  }
+
+  return (
+    <StyledAlert severity="error" onClose={onClose}>
+      <MuiAlertTitle>
+        This safe is using an invalid master copy contract. The web interface might not work correctly. It&apos;s
+        recommended to use the{' '}
```

Comment:
```suggestion
        This Safe is created with an unsupported base contract. The web interface might not work correctly. We recommend using the{' '}
```

Reply:
The ecosystem term for the "base" contract is an implementation contract. But in our codebase we use "singleton", I'd suggest to keep the wording aligned

Reply:
Thanks for the correction Mikhail. IMHO implementation contract is clear, singleton sounds like too much detail. 
Let's just drop the word `base` maybe?
I.e. `This Safe was created with an unsupported contract`

Reply:
> This Safe was created with an unsupported contract

It may be too broad, I feel like creating a Safe with an unsupported factory contract also can be described like this

Reply:
Yes, and that would also warrant this banner.